x/slashing/src/abci_handler.rs

```diff
@@ -26,4 +27,31 @@ impl<SK: StoreKey, PSK: ParamsSubspaceKey, SSK: SlashingStakingKeeper<SK, M>, M:
     pub fn genesis<DB: Database>(&self, ctx: &mut InitContext<'_, DB, SK>, genesis: GenesisState) {
         self.keeper.init_genesis(ctx, genesis)
     }
+
+    /// begin_block check for infraction evidence or downtime of validators
+    /// on every begin block
+    pub fn begin_block<DB: Database>(
+        &self,
+        ctx: &mut BlockContext<'_, DB, SK>,
+        request: RequestBeginBlock,
+    ) {
+        // Iterate over all the validators which *should* have signed this block
+        // store whether or not they have actually signed it and slash/unbond any
+        // which have missed too many blocks in a row (downtime slashing)
+        if let Some(vote_info) = request.last_commit_info {
+            for vote in vote_info.votes {
+                self.keeper
+                    .handle_validator_signature(
+                        ctx,
+                        vote.validator.address.into(),
+                        vote.validator.power as u32,
+                        vote.signed_last_block,
+                    )
+                    .expect(
+                        "method `handle_validator_signature` is called from infallible method.
```

Comment:
We try to follow the advice given [here](https://doc.rust-lang.org/std/result/enum.Result.html) on how to use expect: "We recommend that expect messages are used to describe the reason you expect the `Result` should be `Ok`". I understand that not all projects use `expect` this way but I think we should. In which case you'll have to explicitly panic here - assuming that's how cosmos handles this situation.